test/Spaces/terrain_warp.jl

```diff
@@ -56,22 +56,42 @@ function warpedspace_2D(
     vert_face_space = Spaces.FaceFiniteDifferenceSpace(vertmesh)
 
     # Generate Horizontal Space
-    horzdomain = Domains.IntervalDomain(
-        Geometry.XPoint{FT}(xlim[1]),
-        Geometry.XPoint{FT}(xlim[2]);
-        periodic = true,
-    )
-    horzmesh = Meshes.IntervalMesh(horzdomain; nelems = helem)
-    horztopology = Topologies.IntervalTopology(horzmesh)
     quad = Spaces.Quadratures.GLL{npoly + 1}()
-    hspace = Spaces.SpectralElementSpace1D(horztopology, quad)
-
+    if ndims == 2
+        horzdomain = Domains.IntervalDomain(
+            Geometry.XPoint{FT}(xlim[1]),
+            Geometry.XPoint{FT}(xlim[2]);
+            periodic = true,
+        )
+        horzmesh = Meshes.IntervalMesh(horzdomain; nelems = helem)
+        horztopology = Topologies.IntervalTopology(horzmesh)
+        hspace = Spaces.SpectralElementSpace1D(horztopology, quad)
+    elseif ndims == 3
+        horzdomain = Domains.RectangleDomain(
+            Geometry.XPoint{FT}(xlim[1]) .. Geometry.XPoint{FT}(xlim[2]),
+            Geometry.YPoint{FT}(xlim[1]) .. Geometry.YPoint{FT}(xlim[2]),
+            x1periodic = true,
+            x2periodic = true,
+        )
+        horzmesh = Meshes.RectilinearMesh(horzdomain, helem, helem)
+        horztopology = Topologies.Topology2D(horzmesh)
```

Comment:
Should we add a ClimaComms context here in anticipation that we'll need to test this on the GPU?

Reply:
(`Topologies.Topology2D(comms_ctx, mesh)`)

Reply:
Yes - I can update this all instances of `Topologies.Topology2D` in this file.

Reply:
Updated.